amplify/backend/function/fetchDomainBalance/src/services/tokens.js

```diff
@@ -7,18 +7,28 @@ const getConvertedTokenAmount = (
   decimals,
   exchangeRate,
 ) => {
-  const tokenScaleFactor = BigNumber.from(10).pow(decimals);
+  // Number of decimals for exchange rate precision
   const exchangeRateDecimals = 9;
+  // Scale factor for exchange rate precision
+  const exchangeRateScaleFactor = BigNumber.from(10).pow(exchangeRateDecimals);
+  // Scale factor based on token's decimals
+  const tokenScaleFactor = BigNumber.from(10).pow(decimals);
+  // Convert exchange rate to BigNumber and normalize it with respect to token scale.
+  // We multiply exchange rate by 10^exchangeRateDecimals to remove decimals, then adjust it by token scale.
   const normalizedTokenExchangeRate = BigNumber.from(
     Math.round(exchangeRate * Math.pow(10, exchangeRateDecimals)),
-  );
+  )
+    .mul(tokenScaleFactor)
+    .div(exchangeRateScaleFactor);
   const formattedNetworkFee = BigNumber.from(networkFee || 0);
 
+  // Calculate final amount considering network fees and exchange rate, normalizing to 10^18 scale
   return BigNumber.from(amount)
-    .add(formattedNetworkFee)
-    .div(tokenScaleFactor)
-    .mul(normalizedTokenExchangeRate)
-    .div(BigNumber.from(10).pow(exchangeRateDecimals));
+    .add(formattedNetworkFee) // Add network fee to the initial amount
+    .mul(normalizedTokenExchangeRate) // Adjust amount using the normalized exchange rate
+    .mul(BigNumber.from(10).pow(18)) // Scale the result to 10^18 format for consistency
```

Comment:
Hardcoding this to `18` seems like a potential issue to me. I'm not sure if this should be linked to the token decimals instead?

Reply:
What I tried to do is to normalise all amounts to `10^18` no matter the used token, so that the total amount will always contain `18` extra decimals